### PR TITLE
Make sure container public port value is not None

### PR DESCRIPTION
**Pull request by kitchoi**
Fix bug described in #63

### CODE REVIEW

File: remoteappmanager/docker/container.py [hunk 1]
```diff
@@ -78,10 +78,10 @@ def from_docker_containers_dict(cls, docker_dict):
         """
         if docker_dict.get('Ports'):
             ip = docker_dict['Ports'][0].get('IP', "")
-            port = docker_dict['Ports'][0].get('PublicPort')
+            port = docker_dict['Ports'][0].get('PublicPort', 80)
         else:
             ip = ""
```
Comment (stefanoborini) [inline remoteappmanager/docker/container.py]: Which brings me to the question... should we even fill this information at all? The defaults are already there...

Comment (kitchoi) [inline remoteappmanager/docker/container.py]: Well, I can do this differently

**Comment by itziakos** [comment]
Can this be merged?

**Comment by stefanoborini** [comment]
@itziakos I think it's redundant to use the default_value. The trait already has it, so it's just a matter of not doing anything, but I might be wrong.

**Comment by kitchoi** [comment]
Alternatively one can construct a set of keyword arguments to pass to the class.  It is the same...

**Comment by kitchoi** [comment]
@sbo Do you prefer this?:
```
        kwargs = {}

        if 'Id' in docker_dict:
            kwargs['docker_id'] = docker_dict['Id']

        if 'Name' in docker_dict:
            kwargs['name'] = docker_dict['Name']

        if 'Image' in docker_dict:
            kwargs['image_name'] = docker_dict['Image']

        if 'ImageID' in docker_dict:
            kwargs['image_id'] = docker_dict['ImageID']

        if docker_dict.get('Ports'):
            if 'IP' in docker_dict['Ports'][0]:
                kwargs['ip'] = docker_dict['Ports'][0]['IP']

            if 'PublicPort' in docker_dict['Ports'][0]:
                kwargs['port'] = docker_dict['Ports'][0]['PublicPort']

        labels = docker_dict.get("Labels", {})

        if SIMPHONY_NS+'mapping_id' in labels:
            kwargs['mapping_id'] = labels[SIMPHONY_NS+"mapping_id"]

        if SIMPHONY_NS+'url_id' in labels:
            kwargs['url_id'] = labels[SIMPHONY_NS+"url_id"]

        return cls(**kwargs)
```

**Comment by kitchoi** [comment]
Or create the class first and then assign values to attributes... There are so many ways...

**Comment by stefanoborini** [comment]
@kitchoi no, you instantiate cls() empty, and then fill the stuff in.

**Comment by kitchoi** [comment]
@sbo That's what I mean by "Or create the instance first and then assign values to attributes..."

**Comment by kitchoi** [comment]
@sbo
```
        instance = cls()

        if 'Id' in docker_dict:
            instance.docker_id = docker_dict['Id']

        if 'Name' in docker_dict:
            instance.name = docker_dict['Name']

        if 'Image' in docker_dict:
            instance.image_name = docker_dict['Image']

        if 'ImageID' in docker_dict:
            instance.image_id = docker_dict['ImageID']

        if docker_dict.get('Ports'):
            if 'IP' in docker_dict['Ports'][0]:
                instance.ip = docker_dict['Ports'][0]['IP']

            if 'PublicPort' in docker_dict['Ports'][0]:
                instance.port = docker_dict['Ports'][0]['PublicPort']

        labels = docker_dict.get("Labels", {})

        if SIMPHONY_NS+'mapping_id' in labels:
            instance.mapping_id = labels[SIMPHONY_NS+"mapping_id"]

        if SIMPHONY_NS+'url_id' in labels:
            instance.url_id = labels[SIMPHONY_NS+"url_id"]

        return instance
```

**Comment by itziakos** [comment]
please sync with master and merge

**Pull request merged**